### PR TITLE
feat(indy-sdk): replace indy SDK with AFJ

### PR DESCRIPTION
**Pull request by outSH**
- Refactor test ledger `indy-testnet` into `indy-all-in-one`. New package uses
    the latest indy version, has healtcheck script, updated startup / cleanup
    scripts.
- Remove `indy-sdk-cli` image since it's not used anymore.
- Refactor `cactus-example-discounted-asset-trade` to use own aries agent
    instead of indy connector. This way it doesn't need to use indy-sdk anymore,
    and python indy connector can be safely removed / upgraded.
- Update sample app readme to explain current workflow.
- Remove client scripts since `cactus-example-discounted-asset-trade-client`
    can now be used to interact with the sample app.
- Add `cactus-example-discounted-asset-trade-client`. It contains script for
    setting up test credentials on the ledger, script with interactive menu for
    interacting with `cactus-example-discounted-asset-trade` sample app,
    and bunch of helper functions used for writing these apps.

Depends on #2859
Depends on #2860

Signed-off-by: Michal Bajer <michal.bajer@fujitsu.com>

**Pull Request Requirements**
- [ ] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [ ] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [ ] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [ ] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [ ] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

File: packages/cactus-plugin-ledger-connector-iroha/package.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
     "axios": "1.5.1",
     "express": "4.18.2",
     "fast-safe-stringify": "2.1.1",
+    "grpc": "1.24.11",
```
Comment (outSH) [inline packages/cactus-plugin-ledger-connector-iroha/package.json:66]: Without it the sample app can't be started, will remove it just before merge (kept for now to simplify manual testing)

**Review by petermetz — CHANGES_REQUESTED** [review]
@outSH LGTM but please resolve the merge conflicts (fine to do it later when the parent PRs are merged)

File: packages/cactus-plugin-ledger-connector-iroha/package.json (hunk 1 above)
Comment (outSH) [reply packages/cactus-plugin-ledger-connector-iroha/package.json]: Removed

**Review by petermetz — APPROVED** [review]
LGTM

**Review by jagpreetsinghsasan — COMMENTED** [review]
LGTM (much needed feature as Indy SDK just doesn't setup in my environment)
(Already approved by Peter, thus skipping on approving it)

**Review by izuru0 — APPROVED** [review]
LGTM

**Pull request merged**